examples/hybrid/gravitywave_parameterization/gravity_wave_parameterization.jl

```diff
@@ -0,0 +1,228 @@
+function gravity_wave_cache(
+    ::Type{FT};
+    source_height = FT(15000),
+    Bm = FT(1.2),
+    F_S0 = FT(4e-3),
+    dc = FT(0.6),
+    cmax = FT(99.6),
+    c0 = FT(0),
+    kwv = FT(2π/100e5),
+    cw = FT(40.0),
+
+) where {FT}
+
+    nc = Int(floor(FT(2 * cmax / dc + 1)))
+    c = [FT((n - 1) * dc - cmax) for n in 1:nc]
+
+    return (;
+        gw_source_height = source_height,
+        gw_F_S0 = F_S0,
+        gw_Bm = Bm,
+        gw_c = c,
+        gw_cw = cw,
+        gw_c0 = c0,
+        gw_nk = length(kwv),
+        gw_k = kwv,
+        gw_k2 = kwv.^2,
+    )
+end
+
+
+
+function gravity_wave_tendency!(Yₜ, Y, p, t)
+    #unpack
+    (; gw_source_height, gw_Bm, gw_F_S0, gw_c, gw_cw, gw_c0, gw_nk, gw_k, gw_k2) = p
+    (; ᶜts, ᶜT, params) = p
+    ᶜρ = Y.c.ρ
+    # parameters
+    thermo_params = CAP.thermodynamics_params(params)
+    grav = FT(CAP.grav(params))
+    cp_m = @. TD.cp_m(thermo_params, ᶜts)
+
+    # compute buoyancy frequency
+    @. ᶜT = TD.air_temperature(thermo_params, ᶜts)
+
+    grad_scalar = similar(ᶜT)
+    parent(grad_scalar) .= parent(Geometry.WVector.(ᶜgradᵥ.(ᶠinterp.(ᶜT))))
+
+    ᶜbf = @. (grav / ᶜT) * (grad_scalar + grav / cp_m)
+    ᶜbf = @. ifelse(ᶜbf < FT(2.5e-5), FT(sqrt(2.5e-5)), sqrt(abs(ᶜbf))) # to avoid small numbers
```

Comment:
Allocating a field, can we inline to avoid allocating or add to the cache to avoid allocating at every `step!`?